File: src/platforms/ios/WangWangPhone/Views/CalculatorApp.swift
import SwiftUI

struct CalculatorAppView: View {
    @Binding var isPresented: Bool
    @State private var displayText = "0"
    @State private var operand1: Double? = nil
    @State private var operatorSymbol: String? = nil
    @State private var isWaitingForOperand = false
    
    let buttons = [
        ["AC", "+/-", "%", "÷"],
        ["7", "8", "9", "×"],
        ["4", "5", "6", "-"],
        ["1", "2", "3", "+"],
        ["0", ".", "="]
    ]
    
    var body: some View {
        VStack(spacing: 12) {
            Spacer()
            
            // Display
            HStack {
                Spacer()
                Text(displayText)
                    .font(.system(size: 80, weight: .light))
                    .foregroundColor(.white)
                    .lineLimit(1)
                    .minimumScaleFactor(0.5)
            }
            .padding(.horizontal, 24)
            .padding(.bottom, 10)
            
            // Buttons Grid
            ForEach(buttons, id: \.self) { row in
                HStack(spacing: 12) {
                    ForEach(row, id: \.self) { label in
                        CalculatorButton(
                            label: label,
                            backgroundColor: buttonBackgroundColor(label),
                            foregroundColor: buttonForegroundColor(label),
                            isWide: label == "0"
                        ) {
                            buttonTapped(label)
                        }
                    }
                }
            }
        }
        .padding(.horizontal, 12)
        .padding(.bottom, 40)
        .background(Color.black)
        .overlay(
            VStack {
                HStack {
                    Button(action: { isPresented = false }) {
                        Image(systemName: "chevron.down")
                            .font(.system(size: 20, weight: .bold))
                            .foregroundColor(.white.opacity(0.6))
                            .padding()
                    }
                    Spacer()
                }
                Spacer()
            }
        )
    }
    
    private func buttonBackgroundColor(_ label: String) -> Color {
        if ["÷", "×", "-", "+", "="].contains(label) {
            return Color.orange
        } else if ["AC", "+/-", "%"].contains(label) {
            return Color(white: 0.65)
        } else {
            return Color(white: 0.2)
        }
    }
    
    private func buttonForegroundColor(_ label: String) -> Color {
        if ["AC", "+/-", "%"].contains(label) {
            return .black
        } else {
            return .white
        }
    }
    
    private func buttonTapped(_ label: String) {
        UIImpactFeedbackGenerator(style: .light).impactOccurred()
        
        switch label {
        case "0"..."9":
            if isWaitingForOperand || displayText == "0" {
                displayText = label
                isWaitingForOperand = false
            } else {
                displayText += label
            }
        case ".":
            if isWaitingForOperand {
                displayText = "0."
                isWaitingForOperand = false
            } else if !displayText.contains(".") {
                displayText += "."
            }
        case "AC":
            displayText = "0"
            operand1 = nil
            operatorSymbol = nil
            isWaitingForOperand = false
        case "+", "-", "×", "÷":
            let currentVal = Double(displayText) ?? 0
            if operand1 == nil {
                operand1 = currentVal
            } else if let op = operatorSymbol, !isWaitingForOperand {
                let result = performCalculation(operand1!, currentVal, op)
                displayText = formatResult(result)
                operand1 = result
            }
            operatorSymbol = label
            isWaitingForOperand = true
        case "=":
            let currentVal = Double(displayText) ?? 0
            if let op1 = operand1, let op = operatorSymbol {
                let result = performCalculation(op1, currentVal, op)
                displayText = formatResult(result)
                operand1 = nil
                operatorSymbol = nil
                isWaitingForOperand = true
            }
        case "+/-":
            if let val = Double(displayText) {
                displayText = formatResult(val * -1)
            }
        case "%":
            if let val = Double(displayText) {
                displayText = formatResult(val / 100.0)
            }
        default:
            break
        }
    }
    
    private func performCalculation(_ op1: Double, _ op2: Double, _ symbol: String) -> Double {
        switch symbol {
        case "+": return op1 + op2
        case "-": return op1 - op2
        case "×": return op1 * op2
        case "÷": return op2 != 0 ? op1 / op2 : 0
        default: return op2
        }
    }
    
    private func formatResult(_ result: Double) -> String {
        let formatter = NumberFormatter()
        formatter.minimumFractionDigits = 0
        formatter.maximumFractionDigits = 8
        formatter.numberStyle = .decimal
        formatter.usesGroupingSeparator = false
        return formatter.string(from: NSNumber(value: result)) ?? "\(result)"
    }
}

struct CalculatorButton: View {
    let label: String
    let backgroundColor: Color
    let foregroundColor: Color
    var isWide: Bool = false
    let action: () -> Void
    
    var body: some View {
        Button(action: action) {
            ZStack {
                if isWide {
                    RoundedRectangle(cornerRadius: 45)
                        .fill(backgroundColor)
                        .frame(height: 80)
                } else {
                    Circle()
                        .fill(backgroundColor)
                        .frame(width: 80, height: 80)
                }
                
                Text(label)
                    .font(.system(size: 32, weight: .medium))
                    .foregroundColor(foregroundColor)
                    .padding(.leading, isWide ? 30 : 0)
                    .frame(maxWidth: isWide ? .infinity : 80, alignment: isWide ? .leading : .center)
            }
        }
        .frame(maxWidth: isWide ? .infinity : 80)
    }
}

File: src/platforms/android/app/src/main/java/com/WangWangPhone/ui/CalculatorApp.kt
package com.WangWangPhone.ui

import androidx.activity.compose.BackHandler
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun CalculatorAppScreen(onClose: () -> Unit) {
    var displayText by remember { mutableStateOf("0") }
    var operand1 by remember { mutableStateOf<Double?>(null) }
    var operator by remember { mutableStateOf<String?>(null) }
    var isWaitingForOperand by remember { mutableStateOf(false) }

    BackHandler { onClose() }

    val buttons = listOf(
        listOf("AC", "+/-", "%", "÷"),
        listOf("7", "8", "9", "×"),
        listOf("4", "5", "6", "-"),
        listOf("1", "2", "3", "+"),
        listOf("0", ".", "=")
    )

    fun onNumberClick(number: String) {
        if (isWaitingForOperand) {
            displayText = number
            isWaitingForOperand = false
        } else {
            displayText = if (displayText == "0") number else displayText + number
        }
    }

    fun onOperatorClick(op: String) {
        val currentVal = displayText.toDoubleOrNull() ?: 0.0
        if (operand1 == null) {
            operand1 = currentVal
        } else if (operator != null && !isWaitingForOperand) {
            val result = calculate(operand1!!, currentVal, operator!!)
            displayText = formatResult(result)
            operand1 = result
        }
        operator = op
        isWaitingForOperand = true
    }

    fun onEqualsClick() {
        val currentVal = displayText.toDoubleOrNull() ?: 0.0
        if (operand1 != null && operator != null) {
            val result = calculate(operand1!!, currentVal, operator!!)
            displayText = formatResult(result)
            operand1 = null
            operator = null
            isWaitingForOperand = true
        }
    }

    fun onClearClick() {
        displayText = "0"
        operand1 = null
        operator = null
        isWaitingForOperand = false
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color.Black)
            .statusBarsPadding()
            .padding(bottom = 20.dp)
    ) {
        // Display
        Box(
            modifier = Modifier
                .weight(1f)
                .fillMaxWidth()
                .padding(horizontal = 24.dp),
            contentAlignment = Alignment.BottomEnd
        ) {
            Text(
                text = displayText,
                color = Color.White,
                fontSize = 80.sp,
                fontWeight = FontWeight.Light,
                maxLines = 1
            )
        }

        // Buttons
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 12.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            buttons.forEach { row ->
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    row.forEach { label ->
                        val isOrange = label in listOf("÷", "×", "-", "+", "=")
                        val isGray = label in listOf("AC", "+/-", "%")
                        val weight = if (label == "0") 2f else 1f
                        
                        CalculatorButton(
                            label = label,
                            backgroundColor = when {
                                isOrange -> Color(0xFFF1A33C)
                                isGray -> Color(0xFFA5A5A5)
                                else -> Color(0xFF333333)
                            },
                            contentColor = if (isGray) Color.Black else Color.White,
                            modifier = Modifier.weight(weight),
                            onClick = {
                                when {
                                    label == "AC" -> onClearClick()
                                    label == "=" -> onEqualsClick()
                                    label in listOf("÷", "×", "-", "+") -> onOperatorClick(label)
                                    label == "+/-" -> {
                                        val currentVal = displayText.toDoubleOrNull()
                                        if (currentVal != null) {
                                            displayText = formatResult(currentVal * -1)
                                        }
                                    }
                                    label == "%" -> {
                                        val currentVal = displayText.toDoubleOrNull()
                                        if (currentVal != null) {
                                            displayText = formatResult(currentVal / 100.0)
                                        }
                                    }
                                    label == "." -> {
                                        if (isWaitingForOperand) {
                                            displayText = "0."
                                            isWaitingForOperand = false
                                        } else if (!displayText.contains(".")) {
                                            displayText += "."
                                        }
                                    }
                                    else -> onNumberClick(label)
                                }
                            }
                        )
                    }
                }
            }
        }
        
        // Home Indicator
        Spacer(modifier = Modifier.height(20.dp))
        Box(modifier = Modifier
            .width(120.dp).height(5.dp).clip(CircleShape)
            .background(Color.White.copy(alpha = 0.8f))
            .align(Alignment.CenterHorizontally))
    }
}

@Composable
fun CalculatorButton(
    label: String,
    backgroundColor: Color,
    contentColor: Color,
    modifier: Modifier = Modifier,
    onClick: () -> Unit
) {
    Box(
        modifier = modifier
            .aspectRatio(if (label == "0") 2.2f else 1f)
            .clip(CircleShape)
            .background(backgroundColor)
            .clickable { onClick() },
        contentAlignment = Alignment.Center
    ) {
        Text(
            text = label,
            color = contentColor,
            fontSize = 32.sp,
            fontWeight = FontWeight.Medium
        )
    }
}

private fun calculate(op1: Double, op2: Double, operator: String): Double {
    return when (operator) {
        "+" -> op1 + op2
        "-" -> op1 - op2
        "×" -> op1 * op2
        "÷" -> if (op2 != 0.0) op1 / op2 else 0.0
        else -> op2
    }
}

private fun formatResult(result: Double): String {
    return if (result % 1.0 == 0.0) {
        result.toInt().toString()
    } else {
        String.format("%.8g", result).trimEnd('0').trimEnd('.')
    }
}
